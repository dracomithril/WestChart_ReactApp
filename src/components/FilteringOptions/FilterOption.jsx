/**
 * Created by Gryzli on 23.04.2017.
 */
import React from "react";
import PropTypes from "prop-types";
import {Checkbox} from "react-bootstrap";

export default class FilterOption extends React.Component {
    render() {
        const {store} = this.context;
        const {filters} = store.getState();
        return (<Checkbox {...this.props.control} checked={filters[this.props.input.name].checked}
                          onChange={(e) => {
                              const {id, checked} = e.target;
                              store.dispatch({type: 'TOGGLE_FILTER', id: id, checked: checked})
                          }} >
            {this.props.description.start}
            <input className="num_days" type="number" min={0} step={1}
                   id={this.props.control.id}
                   value={filters[this.props.input.name].days}
                   onChange={(e) => {
                       const target = e.target;
                       const {id, name, value} = target;
                       store.dispatch({
                           type: 'UPDATE_DAYS',
                           id: id,
                           name: name,
                           value: Number(value)
                       })
                   }}/>{this.props.description.end}
        </Checkbox>);
    }
}
FilterOption.contextTypes = {
    store: PropTypes.object
};
FilterOption.propTypes = {};