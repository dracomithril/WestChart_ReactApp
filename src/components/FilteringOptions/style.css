.panel{
    float: left;
    padding: 0;
    margin: 0;
    position: relative;
    z-index: 1000;
}
.panel-heading {
    padding: 10px 5px;
    border-bottom: 1px solid transparent;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
}
.panel-body {
    padding: 5px;
    text-align: left;
}

.num_days {
    padding-left: 5px;
    text-align: right;
    width: 50px;
}