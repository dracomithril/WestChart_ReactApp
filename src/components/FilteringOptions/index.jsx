/**
 * Created by Gryzli on 28.01.2017.
 */
import React from "react";
import PropTypes from 'prop-types';
import {Accordion, Checkbox, OverlayTrigger, Panel, Tooltip} from "react-bootstrap";
import "./FilteringOptions.css";
import filters_def from "./../../filters";
let utils = require('./../../utils');

export default class FilteringOptions extends React.Component {
    render() {
        const {store} = this.context;
        const {filters} = store.getState();
        const toggleAction = (e) => {
            const target = e.target;
            const {id, checked} = target;
            store.dispatch({type: 'TOGGLE_FILTER', id: id, checked: checked})
        };
        return (<Accordion>
                <Panel header="Filters" bsStyle="info">
                    <Checkbox checked={filters['woc_control'].checked} onChange={toggleAction} id="woc">
                        <OverlayTrigger placement="top"
                                        overlay={<Tooltip
                                            id="woc">{`Will show all [${utils.woc_string}]`}</Tooltip>}>
                            <span>
                            [{utils.woc_string}]</span></OverlayTrigger>
                    </Checkbox>
                    <span>Show only posts:</span><br/>
                    <div style={{paddingLeft: 10}}>
                        {filters_def.control.map((elem) => {
                            return (<Checkbox {...elem.control} checked={filters[elem.input.name].checked}
                                              onChange={toggleAction} key={elem.input.name}>
                                {elem.description.start}<input className="num_days" type="number" min={0} step={1}
                                                               id={elem.control.id}
                                                               value={filters[elem.input.name].days}
                                                               onChange={(e) => {
                                                                   const target = e.target;
                                                                   const {id, name, value} = target;
                                                                   store.dispatch({
                                                                       type: 'UPDATE_DAYS',
                                                                       id: id,
                                                                       name: name,
                                                                       value: Number(value)
                                                                   })
                                                               }}/>{elem.description.end}
                            </Checkbox>)
                        })}
                    </div>
                </Panel>
            </Accordion>
        )
    }
}
FilteringOptions.contextTypes = {
    store: PropTypes.object
}
FilteringOptions.propTypes = {
    state: PropTypes.object,
    onChange: PropTypes.func,
    woc_string: PropTypes.string
};