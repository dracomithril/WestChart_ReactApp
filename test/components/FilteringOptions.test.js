/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import FilteringOptions from "../../src/front/components/FilteringOptions/index";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import configureMockStore from "redux-mock-store";
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<FilteringOptions/>', () => {
    it('renders without crashing ChartPresenter', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <FilteringOptions />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    xit('should be able to change state of woc checkbox', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <FilteringOptions />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        let btn = wrapper.findWhere(n => n.props().id === 'bFilters').first();
        btn.simulate('click');
        let findWhere = wrapper.findWhere(n => n.props().id === 'woc_cb');
        const wocCheckbox = findWhere.first();
        let woc_props = wocCheckbox.props();
        expect(woc_props.checked).toBeTruthy();
        wocCheckbox.simulate('change',{target:{id:'woc',checked:false}});
        let actions = store.getActions();
        expect(actions[0].type).toBe('TOGGLE_FILTER');
        expect(actions[0].id).toBe('woc');
        expect(actions[0].checked).toBeFalsy();
    })

});
