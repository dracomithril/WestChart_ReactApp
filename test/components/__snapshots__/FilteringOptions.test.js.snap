// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilteringOptions/> renders without crashing ChartPresenter 1`] = `
<Accordion>
  <Panel
    bsClass="panel"
    bsStyle="info"
    defaultExpanded={false}
    header="Filters"
  >
    <Checkbox
      bsClass="checkbox"
      checked={true}
      disabled={false}
      id="woc"
      inline={false}
      onChange={[Function]}
    >
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="woc"
            placement="right"
          >
            Will show all [Wielkie Ogarnianie Charta]
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <span>
          [
          Wielkie Ogarnianie Charta
          ]
        </span>
      </OverlayTrigger>
    </Checkbox>
    <span>
      Show only posts:
    </span>
    <br />
    <div
      style={
        Object {
          "paddingLeft": 10,
        }
      }
    >
      <FilterOption
        check={[Function]}
        control={
          Object {
            "id": "add",
            "name": "date_added_control",
          }
        }
        description={
          Object {
            "end": " days",
            "start": "added in last ",
          }
        }
        input={
          Object {
            "max": 31,
            "name": "add_control",
          }
        }
        type="countDays"
        valueName="added_time"
      />
      <FilterOption
        check={[Function]}
        control={
          Object {
            "id": "create",
            "name": "date_create_control",
          }
        }
        description={
          Object {
            "end": " days",
            "start": "created in last ",
          }
        }
        input={
          Object {
            "max": 31,
            "name": "create_control",
          }
        }
        type="countDays"
        valueName="created_time"
      />
      <FilterOption
        check={[Function]}
        control={
          Object {
            "id": "update",
            "name": "date_update_control",
          }
        }
        description={
          Object {
            "end": " days",
            "start": "updated in last ",
          }
        }
        input={
          Object {
            "max": 31,
            "name": "update_control",
          }
        }
        type="countDays"
        valueName="updated_time"
      />
      <FilterOption
        check={[Function]}
        control={
          Object {
            "id": "more",
            "name": "more_then_control",
          }
        }
        description={
          Object {
            "start": "more then ",
          }
        }
        input={
          Object {
            "name": "more_control",
          }
        }
        type="more_or_less"
      />
      <FilterOption
        check={[Function]}
        control={
          Object {
            "id": "less",
            "name": "less_then_control",
          }
        }
        description={
          Object {
            "start": "less then ",
          }
        }
        input={
          Object {
            "name": "less_control",
          }
        }
        type="more_or_less"
      />
    </div>
  </Panel>
</Accordion>
`;
